engine/src/main/java/org/destinationsol/systems/DamageSystem.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.systems;
+
+import org.destinationsol.components.Health;
+import org.destinationsol.entitysystem.ComponentSystem;
+import org.destinationsol.entitysystem.RegisterEventReceivers;
+import org.destinationsol.events.DamageEvent;
+import org.terasology.gestalt.entitysystem.entity.EntityRef;
+import org.terasology.gestalt.entitysystem.event.EventResult;
+import org.terasology.gestalt.entitysystem.event.ReceiveEvent;
+
+/**
+ * When a damage event happens to an entity with a health component, this system reads the damage from that event and
+ * lowers its health by that amount. If it would lower the health to less than zero, it's reduced to zero instead. If
+ * the damage is a negative amount, nothing happens.
+ */
+@RegisterEventReceivers
+public class DamageSystem extends ComponentSystem {
```

Comment:
Why are you extending `ComponentSystem`?

Reply:
@Cervator suggested it, because he said that annotations shouldn't fundamentally change anything

Reply:
(And therefore it should extend ComponentSystem too)

Reply:
The only reasons you would want to extend `ComponentSystem` are if you want access to the `context` or you want to tap into the methods it contains. What we were talking about the other day was that the annotation to register the event receiver can be *replaced entirely* with an interface to implement ie. `implements EventReceiver` see [here](https://github.com/NicholasBatesNZ/notepad/commit/4f4445da10f6d2fe29d3daf79dfcd9d5e531ba83).
You've got to undo the changes in [this commit](https://github.com/MovingBlocks/DestinationSol/pull/496/commits/006da6e2778bb4b0bded428b5f953387f42666ff) to do that.
Ping me on discord if you'd like more clarification :+1:

Reply:
Thanks 👍
I made the change